src/elm/Models.elm

```diff
@@ -34,16 +38,25 @@ type alias Flags =
 
 initModel : Flags -> Routing.Route -> Model
 initModel flags route =
-    Model route
-        flags.resolverUrl
-        flags.localDomain
-        initUpload
-        initTerms
-        (initTarget flags.dataSourcesIds)
-        initResolver
-        Nothing
-        flags.version
-        Material.model
+    let
+        dataSourceIds =
+            (List.map DataSourceId flags.dataSourcesIds)
+
+        initialTarget =
+            Target.initial
+    in
+        Model route
+            NotAsked
+            dataSourceIds
+            flags.resolverUrl
+            flags.localDomain
+            initUpload
+            initTerms
+            ({ initialTarget | current = List.head dataSourceIds })
+            initResolver
+            Nothing
+            flags.version
+            Material.model
```

Comment:
I think I'd rather have an explicit record here than a function call with this many arguments.